src/overlays/actors/ovl_T_Keitai/ac_t_keitai.c

```diff
@@ -2,42 +2,199 @@
+    f32 currentFrame = this->skeletonInfo.frameControl.currentFrame;
+    f32 startingChno = start_chno[index];
+    f32 endingChno = end_chno[index];
+    f32 startingScale = start_scale[index];
+    f32 endingScale = end_scale[index];
+    f32 scale;
+
+    if (currentFrame < startingChno) {
+        scale = startingScale;
+    } else if (endingChno < currentFrame) {
+        scale = endingScale;
+    } else {
+        scale = (((currentFrame - startingChno) / (endingChno - startingChno)) * (endingScale - startingScale)) +
+                startingScale;
+    }
+
+    this->toolActor.actor.scale.x = scale;
+    this->toolActor.actor.scale.y = scale;
+    this->toolActor.actor.scale.z = scale;
+}
+
+void aTKT_takeout(T_Keitai* this) {
+    aTKT_calc_scale(this, 0);
+}
+
+void aTKT_putaway(T_Keitai* this) {
+    aTKT_calc_scale(this, 1);
+}
+
+void aTKT_destruct(T_Keitai* this) {
+    Actor_delete(&this->toolActor.actor);
+}
+
+void aTKT_setupAction(T_Keitai* this, s32 processIndex) {
+    static T_KeitaiActionFunc process[] = {
+        aTKT_takeout, aTKT_putaway, aTKT_destruct, (T_KeitaiActionFunc)(void*)none_proc1, NULL,
+    };
+    this->process = process[processIndex];
+    this->processIndex = processIndex;
+
+    {
+        T_Keitai_Animation_Data* animData = &aTKT_anm_dt[processIndex];
+
+        cKF_SkeletonInfo_R_init(&this->skeletonInfo, this->skeletonInfo.skeleton, animData->animation,
+                                animData->startFrame, animData->endFrame, animData->startFrame, 1.0f, 0.0f,
+                                ANIMATION_STOP, NULL);
+    }
+}
+
+void aTKT_actor_move(Actor* thisx, Game_Play* game_play) {
+    T_Keitai* this = THIS;
+
+    func_80A1F58C_jp(this, game_play);
+
+    if (this->toolActor.unk1BC != this->processIndex) {
+        aTKT_setupAction(this, this->toolActor.unk1BC);
+    }
+
+    cKF_SkeletonInfo_R_play(&this->skeletonInfo);
+    this->process(this);
+}
+
+#define AC_GCN_OPEN_DISP(gfxCtx)            \
+    {                                       \
+        GraphicsContext* __gfxCtx = gfxCtx; \
+        int __gfx_opened = 0;               \
+        do {                                \
+        } while (0)
+
+#define AC_GCN_CLOSE_DISP(gfxCtx) \
+    (void)__gfx_opened;           \
+    }                             \
+    do {                          \
+    } while (0)
+
+#define OPEN_POLY_OPA_DISPS()                 \
+    {                                         \
+        Gfx* __polyOpa = __gfxCtx->polyOpa.p; \
+        int __opa_opened = 0;                 \
+        do {                                  \
+        } while (0)
+
+#define CLOSE_POLY_OPA_DISPS()       \
+    __gfxCtx->polyOpa.p = __polyOpa; \
+    (void)__opa_opened;              \
+    }                                \
+    do {                             \
+    } while (0)
+
+void aTKT_actor_draw(Actor* thisx, Game_Play* game_play) {
+    static xyz_t D_80A1F9F4_jp = { 0.0f, 0.0f, 0.0f };
```

Comment:
It doesn't seem to appear in any of the maps (DnM+, AC, DnMe+). I think it was replaced with a `Matrix_Position_Zero` call very early in the porting process, because yeah, doing it like this is silly lol